sympy/series/tests/test_limits.py

```diff
@@ -135,6 +135,10 @@ def test_floor():
     assert limit(floor(x), x, 248, "+") == 248
     assert limit(floor(x), x, 248, "-") == 247
 
+    # https://github.com/sympy/sympy/issues/14478
+    assert limit(x/2*floor(3/x), x, 0, '+') == 3/2
```

Comment:
The limit on the left hand size is a SymPy Rational object while 3/2 denotes a Python float. They currently compare equal but that may change in the future (cf. https://github.com/sympy/sympy/pull/20033). Hence it would be preferable to write something like `S(3)/2` on the right hand side. This also applies to the arguments of `AccumBounds` on the next line.

Reply:
Currently,
```py
In [4]: AccumBounds(-0.5, 1.5) == AccumBounds(-S.Half, S(3)/2)
Out[4]: False

In [5]: AccumBounds(-0.5, 1.5) == AccumBounds(Rational(-1, 2), Rational(3, 2))
Out[5]: False

In [6]: limit(-floor(x/2) + floor(x/2 + 1/2), x, oo) == AccumBounds(Rational(-1, 2), Rational(3, 2))
Out[6]: False
```
Is this a bug in AccumBounds?

Reply:
I don't think so. It seems that `Expr.__eq__` works that way with numbers that are embedded in an expression: https://github.com/sympy/sympy/blob/318376af79c245c49114e67bf590aa716ac7091b/sympy/core/expr.py#L151-L156
`-0.5` and `-S.Half` currently compare True but they are of different type and therefore `AccumBound` objects with such parameters will compare False.